### ISSUE TITLE
Privacy plugin seems having problems with @ symbol in filename

### ISSUE THREAD

**Issue by dvonessen** [squidfunk/mkdocs-material#6821]
### Bug description

Hi,

I am unsure if this is a bug or not, therefore I ask here before opening an issue.
I think that the privacy plugin doesn't work with encoded @ -> %2F or any other encoded characters in filenames.

You can see the problem on [gcix.dev](https://gcix.dev/) on all the batches.
There are only two working batches docker and pypi.

Maybe I missed something in the documentation.
Repository: https://gitlab.com/gcix/gcix

Adding some logs too:

```
INFO    -  Documentation built in 10.61 seconds
INFO    -  [16:15:39] Reloading browsers
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/license/gcix%2Fgcix.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/pipeline-status/gcix%2Fgcix.6be10e4e.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/v/tag/gcix%2Fgcix.63de480b.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/last-commit/Gcix%2Fgcix.d72d2dc2.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/contributors/gcix%2Fgcix.63de480b.svg HTTP/1.1" code 404
INFO    -  [16:15:40] Browser connected: http://localhost:8000/
WARNING -  [16:15:40] "GET /versions.json HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/npm/dm/%40gcix/gcix.37b11339.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/matrix/gcix%3Amatrix.4eae8735.org.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/license/gcix%2Fgcix.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/v/tag/gcix%2Fgcix.63de480b.svg HTTP/1.1" code 404
WARNING -  [16:15:40] "GET /assets/external/img.shields.io/gitlab/pipeline-status/gcix%2Fgcix.6be10e4e.svg HTTP/1.1" code 404
```

Thank you for this awesome piece of software.

Daniel

### Related links

- [Link to Discussion](https://github.com/squidfunk/mkdocs-material/discussions/6815)

### Reproduction

INFO    -  Archive successfully created:

  9.5.11-problem-with-privacy-plugin/docs/index.md 448.0 B
  9.5.11-problem-with-privacy-plugin/mkdocs.yml 103.0 B
  9.5.11-problem-with-privacy-plugin/platform.json 62.0 B
  9.5.11-problem-with-privacy-plugin/requirements.lock.txt 320.0 B
  9.5.11-problem-with-privacy-plugin/requirements.txt 31.0 B

  9.5.11-problem-with-privacy-plugin.zip 1.9 kB
[9.5.11-problem-with-privacy-plugin.zip](https://github.com/squidfunk/mkdocs-material/files/14392008/9.5.11-problem-with-privacy-plugin.zip)

### Steps to reproduce

1. Using batches with @ or other special characters, from shields.io
2. Enable `privacy` plugin

### Browser

Chrome

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting and the excellent reproduction. Fixed in ca7e9523f. Indeed this was a shortcoming of the privacy plugin not unquoting URLs before writing the downloaded files to the filesystem. The commit fixes this:

![screenshot-localhost-3000-1708758739444](https://github.com/squidfunk/mkdocs-material/assets/932156/ef423aca-35ea-4657-ab69-dc1e6662a5bf)

**Comment by dvonessen** [comment]
I will wait for the new release. 👍

**Comment by squidfunk** [comment]
Released as part of [9.5.12](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.12).
